cdap-data-fabric/src/main/java/co/cask/cdap/data/stream/service/StreamHandler.java

```diff
@@ -200,8 +208,11 @@ public void create(HttpRequest request, HttpResponder responder,
       // TODO: For create successful, 201 Created should be returned instead of 200.
       responder.sendStatus(HttpResponseStatus.OK);
     } catch (IllegalArgumentException e) {
-      responder.sendString(HttpResponseStatus.BAD_REQUEST,
-                           e.getMessage());
+      LOG.error("Failed to create stream {}", stream, e);
+      responder.sendString(HttpResponseStatus.BAD_REQUEST, e.getMessage());
+    } catch (NotFoundException e) {
+      LOG.error("Failed to create stream {}", stream, e);
+      responder.sendString(HttpResponseStatus.BAD_REQUEST, e.getMessage());
```

Comment:
I added a catch(Exception) at the end, but having the other exceptions higher up means many more try catch blocks, which makes the code messy without too much benefit, imo.